homeassistant/components/starlink/sensor.py

```diff
@@ -0,0 +1,77 @@
+"""Contains sensors exposed by the Starlink integration."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorStateClass
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import DATA_RATE_MEGABITS_PER_SECOND, DEGREE, TIME_MILLISECONDS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import StarlinkSensorEntity, StarlinkSensorEntityDescription
+
+SENSORS: tuple[StarlinkSensorEntityDescription, ...] = (
+    StarlinkSensorEntityDescription(
+        key="ping",
+        name="Ping",
+        icon="mdi:speedometer",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=TIME_MILLISECONDS,
+        value_fn=lambda data: round(data.pop_ping_latency_ms),
+    ),
+    StarlinkSensorEntityDescription(
+        key="azimuth",
+        name="Azimuth",
+        icon="mdi:compass",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=DEGREE,
+        value_fn=lambda data: data.direction_azimuth,
+    ),
+    StarlinkSensorEntityDescription(
+        key="elevation",
+        name="Elevation",
+        icon="mdi:compass",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=DEGREE,
+        value_fn=lambda data: data.direction_elevation,
+    ),
+    StarlinkSensorEntityDescription(
+        key="uplink_throughput",
+        name="Uplink throughput",
+        icon="mdi:upload",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=DATA_RATE_MEGABITS_PER_SECOND,
+        value_fn=lambda data: round(data.uplink_throughput_bps / 1000000, 2),
+    ),
+    StarlinkSensorEntityDescription(
+        key="downlink_throughput",
+        name="Downlink throughput",
+        icon="mdi:download",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=DATA_RATE_MEGABITS_PER_SECOND,
+        value_fn=lambda data: round(data.downlink_throughput_bps / 1000000, 2),
+    ),
+    StarlinkSensorEntityDescription(
+        key="last_boot_time",
+        name="Last boot time",
+        icon="mdi:download",
+        state_class=SensorStateClass.MEASUREMENT,
```

Comment:
I believe, this sensor does not fit the [definition of measurement](https://developers.home-assistant.io/docs/core/entity/sensor/#available-state-classes) and long term stats don't make sense for it. Let's remove this line.

Reply:
Good catch! Looks like I missed some stuff while copy-pasting